### ISSUE TITLE
A few questions

### ISSUE THREAD

**Issue by dylanirlbeck** [tylershuster/urbit#7]
Hey! I'm not sure if my DMs are going through, but here are a few questions I've come across.

The context is that I'm trying to build a backend that consists of a Gall app and a Node.js server. I'd like the Node server to send data to the Gall app, have Gall perform some computation, and return that data back to the server.

1. Which method(s) would I want to use in your library to accomplish this? 
2. Moreover, how would I set up the Gall app properly? I'm relatively new to Urbit so these things don't seem super straightforward.

Historically I've only played around with poking Gall apps but have never need to return data from them, only print things. I'm wondering if I'll need to learn about Gall subscriptions in order to pass data back to the caller (in this case, the Node server). Thanks so much in advance!

**Comment by dylanirlbeck** [comment]
A nice tutorial/example piece of code would be receiving data from a subscription and just `console.log`ing it -- the current examples only seem to set up the subscription and nothing else.

**Comment by tylershuster** [comment]
Yeah the DM I initiated seems to have disappeared. I'm certainly not getting them.

You may want to take a look at https://github.com/tylershuster/urbit/blob/master/example/index.html

Once you've downloaded the repo, you can just start that and follow the instructions and the codeblock will output all the stuff that goes into the console.

One thing you can do is like this:
```js
const airlock = Airlock.authenticate(...);
airlock.subscribe(app, path, { event: (data) => console.log(data) });
```
You can make a subscription with a handler named `event` that will fire with data every time that subscription receives something.

You can also use the magic method like so:
```js
// Say you're expecting a subscription message like { 'graph-store-update' : {...} }
airlock.onGraphStoreUpdate(data => handleData(data));
```

The class accepts any property beginning with `on`, then transforms camelCase into kebab-case and looks for messages matching that data.


***

That's it as far as the JavaScript side of it goes. As far as setting up the gall side of it goes, I am not the man to ask. If this is about `ucal`, I would ask Josh and the others something along the lines of "what data exists at which paths?" (and of course, what's the app name?). Then you'll be able to do stuff like `airlock.subscribe('ucal', '/calendars', { event: cal => handleCal(cal) })`

Close this issue if it works out for you and if not try deleting the DM with me and starting over again. If that still doesn't work, we can keep this thread going.
